addons/website_sale/models/sale_order.py

```diff
@@ -174,6 +187,28 @@ def _cart_accessories(self):
             accessory_products -= order.website_order_line.mapped('product_id')
             return random.sample(accessory_products, len(accessory_products))
 
+    @api.multi
+    def action_recovery_email_send(self):
+        compose_form_id = self.env.ref('mail.email_compose_message_wizard_form').id
+        template_id = self.env.ref('website_sale.recovery_mail').id
+        ctx = {
+            'default_composition_mode': 'mass_mail' if len(self) > 1 else 'comment',
+            'default_res_id': self.ids[0],
+            'default_model': 'sale.order',
+            'default_use_template': bool(template_id),
```

Comment:
This is strange. If the template does not exists you method will crash at template_id = self.env.ref('website_sale.recovery_mail').id because ref either crashes either returns None when the xml_id is not present anymore.